test/testEditor.ts

```diff
@@ -102,18 +104,32 @@ export const TestEditor = {
     );
   },
 
-  /**
-   * Insert the given set of slate nodes at the given path.
+  /** 
+   * Apply one TransformFunc to slate.
+   */
+  applyTransform: async (e: TestEditor, transform: TransformFunc): Promise<unknown> => {
+    return new Promise((resolve, reject) => {
+      e.onChangeComplete = () => {
+        e.onChangeComplete = () => void {};
+        resolve();
+      };
+      try {
+        transform(e);
+      } catch (err) {
+        reject(err);
+      }
+    }).catch((err) => {
+      fail(err);
+    });
+  },
+
+  /** 
+   * Apply multiple TransformFuncs to slate.
    */
-  insertSlateNodes: async (e: TestEditor, nodes: Node[], path: number[]) => {
-    await TestEditor.applySlateOpsToSlate(
-      e,
-      nodes.reverse().map((node) => {
-        return {
-          type: 'insert_node',
-          path: path,
-          node: node,
-        };
+  applyTransforms: async (e: TestEditor, transforms: TransformFunc[]) => {
+    await Promise.all(
```

Comment:
`Promise.all` doesn't execute its children serially, could that be a problem here?

Reply:
Yeah, maybe so. And possibly for the other places that I'm using it.

Reply:
Perhaps

  applyTransforms: async (e: TestEditor, transforms: TransformFunc[]) => {
    transforms.map(async (transform) => {
      await TestEditor.applyTransform(e, transform);
    });
  },

is the right answer?